ledger/catchupaccessor.go

```diff
@@ -324,30 +334,39 @@ func (c *CatchpointCatchupAccessorImpl) processStagingBalances(ctx context.Conte
 				for aidx := range accountData.AppParams {
 					err = writeCatchpointStagingCreatable(ctx, tx, balance.Address, basics.CreatableIndex(aidx), basics.AppCreatable)
 					if err != nil {
-						return err
+						return
 					}
 				}
 			}
 
 			hash := accountHashBuilder(balance.Address, accountData, balance.AccountData)
-			added, err := trie.Add(hash)
+			added, err2 := progress.cachedTrie.Add(hash)
```

Comment:
why `err2`? `err` does not look used in parallel here

Reply:
because of the `added`; I didn't want to hide the original err. I will change this by declaring the "added" as bool